src/KK.AspNetCore.EasyAuthAuthentication/Services/EasyAuthForApplicationsService.cs

```diff
@@ -14,29 +14,31 @@ namespace KK.AspNetCore.EasyAuthAuthentication.Services
     using Newtonsoft.Json.Linq;
 
     /// <summary>
-    /// A service that can be used to authentificat applications in the <see cref="EasyAuthAuthenticationHandler"/>.
+    /// A service that can be used to authenticate applications or users in the <see cref="EasyAuthAuthenticationHandler"/>.
+    /// That is for the use case if the user doesn't authenticated on this webapp. So he has only a Authorization Header.
     /// </summary>
-    public class EasyAuthForApplicationsService : IEasyAuthAuthentificationService
+    public class EasyAuthForAuthorizationTokenService : IEasyAuthAuthentificationService
     {
         private const string AuthorizationHeader = "Authorization";
-        private const string JWTIdentifyer = "Bearer";
-        private const string ProviderNameKey = "idp";
+        private const string JWTIdentifier = "Bearer";
+        private const string ProviderNameKey = "idp";        
 
         private static readonly Func<IHeaderDictionary, string, bool> IsHeaderSet =
            (headers, headerName) => !string.IsNullOrEmpty(headers[headerName].ToString());
 
-        private readonly ProviderOptions defaultOptions = new ProviderOptions(typeof(EasyAuthForApplicationsService).Name)
+        private readonly ProviderOptions defaultOptions = new ProviderOptions(typeof(EasyAuthForAuthorizationTokenService).Name)
         {
-            NameClaimType = ClaimTypes.Spn
+            NameClaimType = ClaimTypes.Spn,
+            RoleClaimType = "roles"
         };
 
-        private readonly ILogger<EasyAuthForApplicationsService> logger;
+        private readonly ILogger<EasyAuthForAuthorizationTokenService> logger;
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="EasyAuthForApplicationsService"/> class.
+        /// Initializes a new instance of the <see cref="EasyAuthForAuthorizationTokenService"/> class.
         /// </summary>
         /// <param name="logger">The logger for this service.</param>
-        public EasyAuthForApplicationsService(ILogger<EasyAuthForApplicationsService> logger) => this.logger = logger;
+        public EasyAuthForAuthorizationTokenService(ILogger<EasyAuthForAuthorizationTokenService> logger) => this.logger = logger;
 
         /// <inheritdoc/>
         public AuthenticateResult AuthUser(HttpContext context, ProviderOptions options = null)
```

Comment:
tokenJson[ProviderNameKey].ToString() may fail, if idp-claim isn't there. Better implementation would be:
`public AuthenticateResult AuthUser(HttpContext context, ProviderOptions options = null)
        {
            this.defaultOptions.ChangeModel(options);

            var tokenJson = this.GetTokenJson(context.Request.Headers[AuthorizationHeader].FirstOrDefault());
            var claims = this.BuildFromApplicationAuth(tokenJson, this.defaultOptions);

            var identityProviderClaim = tokenJson[IdentityProviderKey];
            if(identityProviderClaim == null)
            {
                /* As stated here (https://docs.microsoft.com/en-us/azure/active-directory/develop/access-tokens), we
                 * can use issuer-claim as identity provider, if idp-claim isn't defined in the token
                 */
                identityProviderClaim = tokenJson[IssuerKey];
            }
            var ticket = AuthenticationTicketBuilder.Build(claims, identityProviderClaim.ToString(), this.defaultOptions);
            return AuthenticateResult.Success(ticket);
        }`

and define as private class member
`private const string IdentityProviderKey = "idp";
        private const string IssuerKey = "iss";`
instead of 
`private const string ProviderNameKey = "idp";`

Reply:
thanks for the input